src/fromager/sdist.py

```diff
@@ -60,21 +60,27 @@ def __init__(
 
 def handle_requirement(
     ctx: context.WorkContext,
-    req: Requirement,
+    original_req: Requirement,
     req_type: str = "toplevel",
     why: list | None = None,
 ) -> str:
     if why is None:
         why = []
-    if not dependencies.evaluate_marker(req):
+    if not dependencies.evaluate_marker(original_req):
         logger.info(
-            f"{req.name}: ignoring {req_type} dependency because of its marker expression"
+            f"{original_req.name}: ignoring {req_type} dependency because of its marker expression"
         )
         return ""
     logger.info(
-        f'{req.name}: {"*" * (len(why) + 1)} handling {req_type} requirement {req} {why}'
+        f'{original_req.name}: {"*" * (len(why) + 1)} handling {req_type} requirement {original_req} {why}'
     )
 
+    req, constraint = ctx.constraints.get_constrained_requirement(original_req)
+    if constraint:
+        logger.info(
+            f"incoming requirement {req} matches constraint {constraint}, the new requirement is {req}"
```

Comment:
```suggestion
            f"{req.name}: incoming requirement {original_req} matches constraint {constraint}, the new requirement is {req}"
```

Prefixing the log message with the requirement name helps keep the console output clear.  I think the other change, to include original_req, was missed in the naming changes.

Reply:
#145